tool/tsh/tshconfig.go

```diff
@@ -46,14 +47,28 @@ type ExtraProxyHeaders struct {
 	Headers map[string]string `yaml:"headers,omitempty"`
 }
 
-func loadConfig(homePath string) (*TshConfig, error) {
-	confPath := filepath.Join(profile.FullProfilePath(homePath), tshConfigPath)
-	configFile, err := os.Open(confPath)
+func loadConfig(fullConfigPath string) (*TshConfig, error) {
+	emptyConfig := &TshConfig{}
+	configFile, err := os.Open(fullConfigPath)
 	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return emptyConfig, nil
+		}
 		return nil, trace.ConvertSystemError(err)
 	}
 	defer configFile.Close()
+
+	bs, err := io.ReadAll(configFile)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	if len(bs) == 0 {
```

Comment:
Nit: I'd use bytes.TrimSpace before comparing the length just in case.

Reply:
I ended up removing the empty/only spaces checks
yaml.Unmarshal handles that just fine

We initially added the check because we were decoding using `yaml.NewDecoder(configFile).Decode(&cfg)` and that requires at least one valid node: https://pkg.go.dev/gopkg.in/yaml.v2#Decoder.Decode
However, the Unmarshal is fine with empty input